src/Tasks/Microsoft.NET.Build.Tasks/NuGetUtils.NuGet.cs

```diff
@@ -47,7 +47,167 @@ public static NuGetFramework ParseFrameworkName(string frameworkName)
             return frameworkName == null ? null : NuGetFramework.Parse(frameworkName);
         }
 
-        public static bool IsApplicableAnalyzer(string file, string projectLanguage)
+        /// <summary>
+        /// Gets a collection of 'analyzer' assets that should be excluded based on the
+        /// <paramref name="compilerApiVersion"/>.
+        /// </summary>
+        /// <remarks>
+        /// This allows packages to ship multiple analyzers that target different versions
+        /// of the compiler. For example, a package may include:
+        ///
+        /// "analyzers/dotnet/roslyn3.7/analyzer.dll"
+        /// "analyzers/dotnet/roslyn3.8/analyzer.dll"
+        /// "analyzers/dotnet/roslyn4.0/analyzer.dll"
+        ///
+        /// When the <paramref name="compilerApiVersion"/> is 'roslyn3.9', only the assets 
+        /// in the folder with the highest applicable compiler version are picked.
+        /// In this case,
+        /// 
+        /// "analyzers/dotnet/roslyn3.8/analyzer.dll"
```

Comment:
I don't think so, if the Roslyn compiler version is 3.9, then it can't necessarily run analyzers written for Roslyn 4.0, so it should use the ones written for Roslyn 3.8.

Reply:
Yeah, ignore me I was just having a slow moment.